File: app/src/main/java/br/pedroso/upcomingmovies/di/application/modules/ApplicationModule.java
package br.pedroso.upcomingmovies.di.application.modules;

import android.app.Application;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class ApplicationModule {
    Application application;

    public ApplicationModule(Application application) {
        this.application = application;
    }

    @Provides
    @Singleton
    Application provideApplication() {
        return application;
    }
}

File: app/src/main/kotlin/br/pedroso/upcomingmovies/di/ApplicationModule.kt
package br.pedroso.upcomingmovies.di

import android.app.Application
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class ApplicationModule(var application: Application) {
    @Provides
    @Singleton
    fun provideApplication(): Application {
        return application
    }
}
